mmcls/datasets/pipelines/transforms.py

```diff
@@ -419,7 +419,7 @@ def __call__(self, results):
             num_output_channels = img.shape[2]
             if random.random() < self.gray_prob:
                 if num_output_channels > 1:
-                    img = mmcv.rgb2gray(img)[:, :, None]
+                    img = mmcv.bgr2gray(img, keepdim=True)
```

Comment:
I think we cannot change the behavior directly. The `cvtColor` of OpenCV has different weights of different channels. This modification may introduce BC-breaking changes.
A better method is to add an argument, maybe `color_format`, to change the behavior, and the default value should be `rgb` to keep the original behavior.